### PR TITLE
Add `to_ordinal` feature for ordinal regression/classification

### PR DESCRIPTION
**Pull request by awsaf49**
This PR adds the feature `to_ordinal` for ordinal regression/classification mentioned in issue #17382

### CODE REVIEW

**Comment by awsaf49** [comment]
@gbaned please approve the workflows.

File: keras/utils/np_utils.py
Comment (hertschuh) [inline keras/utils/np_utils.py:97]: Small nitpick: can you move this up line 79 so that it's right after `to_categorical`?

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thank you for the Pull Request!
Just a small change:

File: keras/utils/np_utils.py
Comment (haifeng-jin) [inline keras/utils/np_utils.py:105]: Please use 4 spaces as the indent.

**Review by haifeng-jin — CHANGES_REQUESTED** [review]
Thanks for the PR!
Please also update [this file](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) after the PR is merged, by adding to the `tf.keras` section for new features.

**Comment by haifeng-jin** [comment]
@awsaf49 Seems the `self.assertTrue()` in the test is failing. Would you please fix it? Thanks.

File: keras/utils/np_utils.py
Comment (awsaf49) [reply keras/utils/np_utils.py]: I think you meant the 2 spaces before `as max(y) + 1` which I converted to 4 spaces in the following commit. If so I think `to_categorical` also uses 2 spaces from which I adopted the format.

Also noticed after `black` formatting it creates a `line too long` error in `fake8` as mentioned in psf/black/issues/1161

File: keras/utils/np_utils.py
Comment (awsaf49) [reply keras/utils/np_utils.py:80]: Moved this in the following commits

File: keras/utils/np_utils.py
Comment (awsaf49) [reply keras/utils/np_utils.py]: > Please also update [this file](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) after the PR is merged, by adding to the tf.keras section for new features.

@haifeng-jin should I create a PR in `tensorflow` for that?

**Comment by awsaf49** [comment]
> @awsaf49 Seems the `self.assertTrue()` in the test is failing. Would you please fix it? Thanks.

`self.assertTrue()` fails for label shape `(3, 2, 1)` as   ordinal->label creates shape `(3, 2)` hence the mismatch. Using `reshape(label)` with `ordinal` before comparison is a fix.

File: keras/utils/np_utils.py
Comment (haifeng-jin) [reply keras/utils/np_utils.py]: The spaces in docstrings are not consistent through the codebase. We try to gradually change them to use 4 spaces. 

In the conflict of black and flake8, we need to manually resolve them.

Yes, you should create a PR in tensorflow afterwards.
Thank you!

**Review by haifeng-jin — APPROVED** [review]
LGTM

**Comment by haifeng-jin** [comment]
@awsaf49 
The tests are still failing.
```
ERROR: test_to_ordinal (__main__.TestNPUtils)
TestNPUtils.test_to_ordinal
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kbuilder/.cache/bazel/_bazel_kbuilder/31d6f47147b75c35404d734345be7323/execroot/org_keras/bazel-out/k8-opt/bin/keras/utils/np_utils_test.runfiles/org_keras/keras/utils/np_utils_test.py", line 74, in test_to_ordinal
    self.assertTrue(
  File "/usr/lib/python3.9/unittest/case.py", line 686, in assertTrue
    if not expr:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

----------------------------------------------------------------------
Ran 3 tests in 0.006s
```

**Comment by awsaf49** [comment]
> @awsaf49 The tests are still failing.
> 
> ```
> ERROR: test_to_ordinal (__main__.TestNPUtils)
> TestNPUtils.test_to_ordinal
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/kbuilder/.cache/bazel/_bazel_kbuilder/31d6f47147b75c35404d734345be7323/execroot/org_keras/bazel-out/k8-opt/bin/keras/utils/np_utils_test.runfiles/org_keras/keras/utils/np_utils_test.py", line 74, in test_to_ordinal
>     self.assertTrue(
>   File "/usr/lib/python3.9/unittest/case.py", line 686, in assertTrue
>     if not expr:
> ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
> 
> ----------------------------------------------------------------------
> Ran 3 tests in 0.006s
> ```

Ah my bad, forgot to put `np.all`, will fix it in the following commit

**Comment by awsaf49** [comment]
@hertschuh I think this PR needs approval from you to be merged.

**Comment by awsaf49** [comment]
This PR is stuck due to
> 1 workflow awaiting approval, First-time contributors need a maintainer to approve running workflows. 

@gbaned  could you please approve?

**Comment by awsaf49** [comment]
I think this PR looks fine but is on hold for quite some time, It would be really helpful if someone could merge the PR. 
cc: @haifeng-jin @hertschuh @gbaned

**Comment by haifeng-jin** [comment]
@awsaf49 I am doing some required Google internal changes since this PR touches the APIs. Sorry for the delay.

**Comment by haifeng-jin** [comment]
@awsaf49 There are more internal review comments that I need to address. So please expect some delays. Thanks.

**Comment by awsaf49** [comment]
> @awsaf49 There are more internal review comments that I need to address. So please expect some delays. Thanks.

In the meantime, I would like to propose creating an example on Ordinal Regression, something like "Age Estimation with Ordinal Regression" on the keras.io/examples page. As of now, there is no mention of Ordinal Regression on the [call for contribution](https://github.com/keras-team/keras-io/blob/master/call_for_contributions.md) page. May I proceed with creating a pull request for this example?

cc: @fchollet

**Comment by awsaf49** [comment]
@haifeng-jin as the PR has been merged, I've created a PR on TensorFlow to update the `RELEASE.md` file as you asked, tensorflow/tensorflow/pull/59436

**Pull request merged**
